### PR TITLE
steps: add version 5.0.1

### PR DESCRIPTION
**Pull request by tristan0x**

### CODE REVIEW

File: var/spack/repos/builtin/packages/steps/package.py [hunk 1]
```diff
@@ -15,19 +17,25 @@ class Steps(CMakePackage):
     maintainers("tristan0x")
 
     license("GPL-3.0-or-later")
+    submodules = True
 
-    version("develop", branch="master", submodules=True)
-    version("4.1.1", submodules=True)
+    version("develop", branch="master")
+    version("5.0.1")
+    version("4.1.1")
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/steps/package.py]: Can we pin the commit sha with `commit=`?

File: var/spack/repos/builtin/packages/steps/package.py [hunk 2]
```diff
@@ -15,19 +17,25 @@ class Steps(CMakePackage):
...
         "codechecks",
         default=False,
         description="Perform additional code checks like code formatting or static analysis",
     )
     variant("lapack", default=False, description="Use new BDSystem/Lapack code for E-Field solver")
     variant(
-        "distmesh", default=True, when="+mpi", description="Add solvers based on distributed mesh"
+        "native",
+        default=False if sys.platform == "darwin" else True,
+        description="Generate non-portable arch-specific code",
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/steps/package.py]: Let's not do this. The target compatibility is encoded in the `target` attribute of specs, and using a variant can create troubles with e.g. reuse. Was there some performance issue behind this change?

**Review by alalazo — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/steps/package.py (hunk 1 above)
Comment (tristan0x) [reply var/spack/repos/builtin/packages/steps/package.py]: sure @alalazo 
I added the SHA in c495bba5b244ae2344a0737403ad56b94197fa97

File: var/spack/repos/builtin/packages/steps/package.py (hunk 2 above)
Comment (tristan0x) [reply var/spack/repos/builtin/packages/steps/package.py]: If I remember correctly, it was to "simplify" the build process for scientists on their MacOS Intel laptops a few years ago. This variant is not needed anymore.

The `target` variant has been removed in c495bba5b244ae2344a0737403ad56b94197fa97

**Review by alalazo — APPROVED** [review]

**Comment by alalazo** [comment]
Thanks!

**Pull request merged**